lib/fs.js

```diff
@@ -381,16 +347,15 @@ fs.access = function(path, mode, callback) {
   if (typeof mode === 'function') {
     callback = mode;
     mode = fs.F_OK;
-  } else if (typeof callback !== 'function') {
-    throw new errors.TypeError('ERR_INVALID_CALLBACK');
   }
 
   path = getPathFromURL(path);
   validatePath(path);
+  maybeCallback(callback);
 
   mode = mode | 0;
   var req = new FSReqWrap();
-  req.oncomplete = makeCallback(callback);
+  req.oncomplete = callback;
```

Comment:
This is the wrong change.  `makeCallback()` returns a function that runs `callback` in the global scope, i.e., with `this === undefined`.  With this change, `this === req`.

Suggestion: get rid of `maybeCallback()` entirely and use `makeCallback()` everywhere.

Reply:
I thought about removing it as well but `makeCallback` has a performance overhead compared to `maybeCallback` due to the wrapping.

I checked all occurrences of `maybeCallback` and found two more places where it wrong right now. I fixed those in a separate commit.

Reply:
Can you put back the makeCallback() here?  It's a needless change now.

Reply:
Done